src/loader/loader.c

```diff
@@ -360,6 +360,22 @@ post_analyze_hook(ParseState *pstate, Query *query)
 		 */
 		switch (nodeTag(query->utilityStmt))
 		{
+			case T_AlterDatabaseStmt:
+			{
+				AlterDatabaseStmt *stmt = (AlterDatabaseStmt *) query->utilityStmt;
+				if (list_length(stmt->options) == 1)
+				{
+					DefElem *option = linitial(stmt->options);
+					if (option->defname && strcmp(option->defname, "tablespace") == 0)
+					{
+						Oid db_oid = get_database_oid(stmt->dbname, false);
+
+						if (OidIsValid(db_oid))
+							ts_bgw_message_send_and_wait(STOP, db_oid);
```

Comment:
should this be a STOP or a RESTART? What starts the bgws up again afterwards. (also should add to test a way to see the the bgws start after this command).

Reply:
i was worried about race conditions with restart but it seems to work fine, i changed alter database rename to RESTART aswell